StepsToCare/CareDashboard.ascx.cs

```diff
@@ -2492,6 +2640,85 @@ private void AddNotificationAttributeControl( bool setValues )
+                {
+                    var noteTemplate = new NoteTemplateService( rockContext ).Get( quickNoteId.Value );
+                    if ( noteTemplate != null )
+                    {
+                        if ( NoConfirm.HasValue && NoConfirm.Value )
+                        {
+                            var noteCreated = createNote( rockContext, careNeedId.Value, noteTemplate.Note, true, noteTemplate, true );
+                            if ( noteCreated )
+                            {
+                                if ( shouldSnoozeNeed )
+                                {
+                                    var careNeed = new CareNeedService( rockContext ).Get( careNeedId.Value );
+                                    if ( careNeed != null && careNeed.CustomFollowUp )
+                                    {
+                                        SnoozeNeed( careNeedId.Value );
+                                    }
+                                }
+                                BindGrid();
+                            }
+                        }
+                        else
+                        {
+                            hfQuickNote_CareNeedId.SetValue( careNeedId.Value );
+                            hfQuickNote_NoteId.SetValue( noteTemplate.Id );
+                            lQuickNote.Text = noteTemplate.Note;
+                            lCareNeedId.Text = careNeedId.ToString();
+                            mdConfirmNote.Show();
+                        }
+                    }
+                }
+
+            }
+        }
+
+        private void SnoozeNeed( int id )
+        {
+            using ( var rockContext = new RockContext() )
+            {
+                var snoozeChildNeeds = GetAttributeValue( AttributeKey.SnoozeChildNeeds ).AsBoolean();
+                var careNeedService = new CareNeedService( rockContext );
+                var careNeed = careNeedService.Get( id );
+                var snoozeValueId = DefinedValueCache.Get( rocks.kfs.StepsToCare.SystemGuid.DefinedValue.CARE_NEED_STATUS_SNOOZED ).Id;
+                if ( careNeed.StatusValueId != snoozeValueId && ( !careNeed.RenewMaxCount.HasValue || careNeed.RenewCurrentCount <= careNeed.RenewMaxCount.Value ) )
+                {
+                    careNeed.StatusValueId = snoozeValueId;
+                    careNeed.SnoozeDate = RockDateTime.Now;
+
+                    if ( snoozeChildNeeds && careNeed.ChildNeeds.Any() )
+                    {
+                        foreach ( var childneed in careNeed.ChildNeeds )
+                        {
+                            childneed.StatusValueId = snoozeValueId;
+                            childneed.SnoozeDate = RockDateTime.Now;
+                        }
+                    }
+                    rockContext.SaveChanges();
+
+                    createNote( rockContext, id, GetAttributeValue( AttributeKey.SnoozeActionText ) );
```

Comment:
Is there a method to compute tense on a string?

Reply:
Unfortunately, no there is not, it is a complicated process that most libraries do not want to handle.